database/database.go

```diff
@@ -67,27 +67,53 @@ func CreateItem[T any](model T) error {
 		return fmt.Errorf("database connection is nil. Make sure to call InitDB() first")
 	}
 
-	result := db.Create(model)
-	if result.Error != nil {
-		return result.Error
+	result := db.Create(model).Error
+	if result != nil {
+		return result
 	}
 	return nil
 }
 
-func GetItems[T any](model *[]T) (*[]T, error) {
+func GetAllItems[T any](model *[]T, column, value string) (*[]T, error) {
 	if db == nil {
 		return nil, fmt.Errorf("database connection is nil. Make sure to call InitDB() first")
 	}
 
-	if err := db.Find(model).Error; err != nil {
+	if err := db.Where(column+"= ?", value).Find(model).Error; err != nil {
 		return nil, err
 	}
 
 	return model, nil
 }
 
-func GetItem[T any](model T) {}
+func GetItem[T any](model *T, column, value string) (*T, error) {
```

Comment:
Have you at least manually tested these generic CRUD methods?

Reply:
yes, i have manually tested them and they all work